src/interactions/slashCommands/SlashCommandOptions.ts

```diff
@@ -0,0 +1,296 @@
+	 */
+	public addIntegerOption(
+		input: SlashCommandIntegerOption | ((builder: SlashCommandIntegerOption) => SlashCommandIntegerOption),
+	) {
+		return this._sharedAddOptionMethod(input, SlashCommandIntegerOption);
+	}
+
+	private _sharedAddOptionMethod<T extends SlashCommandOptionBase>(
+		input: T | ((builder: T) => T),
+		Instance: new () => T,
+	): R {
+		const { options } = this;
+
+		// First, assert options conditions - we cannot have more than 25 options
+		validateMaxOptionsLength(options);
+
+		// Get the final result
+		const result = typeof input === 'function' ? input(new Instance()) : input;
+
+		if (!(result instanceof Instance))
+			throw new TypeError(
+				`Expected to receive a ${Instance.name} builder back, got ${
+					// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+					((result as any)?.name as string) ?? 'Unknown'
+				} ("${typeof result}") instead`,
+			);
+
+		// Push it
+		options.push(result);
+
+		return this as any;
+	}
+}
+
+export class SharedNameAndDescription {
+	protected name!: string;
+	protected description!: string;
+
+	/**
+	 * Sets the name
+	 * @param name The name
+	 */
+	public setName(name: string) {
+		// Assert the name matches the conditions
+		validateName(name);
+
+		this.name = name;
+
+		return this;
+	}
+
+	/**
+	 * Sets the description
+	 * @param description The description
+	 */
+	public setDescription(description: string) {
+		// Assert the description matches the conditions
+		validateDescription(description);
+
+		this.description = description;
+
+		return this;
+	}
+}
+
+// #endregion
+
+// #region Basic options
+
+class SlashCommandOptionBase extends SharedNameAndDescription implements ToAPIApplicationCommandOptions {
+	protected required = false;
+
+	public constructor(protected type: ApplicationCommandOptionType) {
+		super();
+	}
+
+	/**
+	 * Marks the option as required
+	 * @param required If this option should be required
+	 */
+	public setRequired(required: boolean) {
+		// Assert that you actually passed a boolean
+		ow(required, 'required', ow.boolean);
+
+		this.required = required;
+
+		return this;
+	}
+
+	public toJSON() {
+		validateRequiredParameters(this.name, this.description, []);
+
+		// Assert that you actually passed a boolean
+		ow(this.required, 'required', ow.boolean);
+
+		return {
+			type: this.type,
+			name: this.name,
+			description: this.description,
+			required: this.required,
+		};
+	}
+}
+
+// #endregion
+
+// #region Options with choices
+
+abstract class ApplicationCommandOptionWithChoicesBase<T extends string | number>
+	extends SlashCommandOptionBase
+	implements ToAPIApplicationCommandOptions
+{
+	protected choices?: APIApplicationCommandOptionChoice[];
+
+	/**
+	 * Adds a choice for this option
+	 * @param name The name of the choice
+	 * @param value The value of the choice
+	 */
+	public addChoice(name: string, value: T) {
+		if (typeof this.choices === 'undefined') this.choices = [];
```

Comment:
We're using TypeScript, we can use syntax of the future.
```suggestion
		this.choices ??= [];
```